### PR TITLE
*: bump to 4.0

### PR DESCRIPTION
**Pull request by BusyJay**
## What have you changed? (mandatory)

Now that 3.0 is released, master should be updated to another version to be distinguished from the released version. 

## What are the type of the changes? (mandatory)

- Engineering (engineering change which doesn't change any feature or fix any issue)

## How has this PR been tested? (mandatory)

No need.

## Does this PR affect documentation (docs) or release note? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

### CODE REVIEW

**Comment by Connor1996** [comment]
not 4.0?

**Comment by disksing** [comment]
I would like to vote `3.1.0-alpha` instead of `3.1.0-alpha.1`.

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "tikv"
-version = "3.0.0-beta.1"
+version = "4.0.0-alpha"
```
Comment (siddontang) [inline Cargo.toml]: here is 4.0, but your PR title is 3.1

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,8 +3,156 @@ All notable changes to this project are documented in this file.
 See also [TiDB Changelog](https://github.com/pingcap/tidb/blob/master/CHANGELOG.md) and [PD Changelog](https://github.com/pingcap/pd/blob/master/CHANGELOG.md).
 
 ## [Unreleased]
-- Leader responds to learner read index message. (https://github.com/pingcap/raft-rs/pull/220)
-- Snapshot may lose some applied results. (https://github.com/tikv/tikv/pull/4716)
+
+## [3.0.0]
+
++ Engine
+  - Introduce Titan, a key-value plugin that improves write performance for
```
Comment (siddontang) [inline CHANGELOG.md]: I think it is better to add some links to these things.

Comment (BusyJay) [reply CHANGELOG.md]: The file is just checked out from release 3.0 branch. The content was copied from release notes. I think they were linked in the past release already.

File: Cargo.toml (hunk 1 above)
Comment (BusyJay) [reply Cargo.toml]: Updated.

**Pull request merged**